.cbmc-batch/include/proof_helpers/make_common_data_structures.h

```diff
@@ -37,6 +41,41 @@
 #define ASSUME_BOUNDED_ARRAY_LIST(list, max_initial_item_allocation, max_item_size)                                    \
     list = make_bounded_array_list((max_initial_item_allocation), (max_item_size))
 
+struct store_array_list_byte {
+    size_t index;
+    uint8_t byte;
+};
+
+/**
+ * Set of invariants for an aws_array_list structure
+ */
+bool is_valid_array_list(struct aws_array_list *list);
+
+/**
+ * Checks whether the members of an aws_array_list structure are bounded by max_initial_item_allocation and
+ * max_item_size
+ */
+bool is_bounded_array_list(struct aws_array_list *list, size_t max_initial_item_allocation, size_t max_item_size);
+
+/**
+ * Ensures the data member of an aws_array_list structure is correctly allocated
+ */
+void ensure_array_list_has_allocated_data_member(struct aws_array_list *list);
+
+/**
+ * Asserts two aws_array_list structures are equivalent.
+ */
+void assert_array_list_equivalency(
```

Comment:
Its not hard to do this for arbitrary buffers. And then this just becomes a special case of that.